setup.py

```diff
@@ -1,3 +1,19 @@
+import subprocess
+
 import setuptools
 
-setuptools.setup()
+
+def get_version():
+    try:
+        return (
+            subprocess.check_output(["git", "describe"])
+            .decode("utf-8")
+            .strip()
+            .replace("-", "+", 1)
+            .replace("-", ".")
+        )
```

Comment:
This should ensure we have a PEP 440 valid version number

Reply:
Wait, doesn't this mean that we can't install the project with a meaningful version from pip without using the wheel?

Reply:
Oh, it had completely slipped my mind that this code is executed at install time for sdists :o you're right!

Let me give it a thought, I'll make a new proposal.

Reply:
@k4nar I've added a new commit for that.

I've added a dynamic VERSION.txt file at release time, which gets removed right after release. Being present in the sdist (and incidentally in the whell too, but too meh), it's read at install time.

I'm not a huge fan, ...
... but my life as a wind turbine would probably have been very different. ;)

Reply:
Works for me :) .